proto/tendermint/abci/types.proto

```diff
@@ -123,23 +123,31 @@ message RequestApplySnapshotChunk {
 }
 
 message RequestPrepareProposal {
-  bytes                   hash   = 1;
-  tendermint.types.Header header = 2 [(gogoproto.nullable) = false];
+  // the modified transactions cannot exceed this size.
+  int64 max_tx_bytes = 1;
   // txs is an array of transactions that will be included in a block,
   // sent to the app for possible modifications.
-  repeated bytes     txs                  = 3;
-  ExtendedCommitInfo local_last_commit    = 4 [(gogoproto.nullable) = false];
-  repeated Evidence  byzantine_validators = 5 [(gogoproto.nullable) = false];
-  // the modified transactions cannot exceed this size.
-  int64 max_tx_bytes = 6;
+  repeated bytes            txs                  = 2;
+  ExtendedCommitInfo        local_last_commit    = 3 [(gogoproto.nullable) = false];
+  repeated Evidence         byzantine_validators = 4 [(gogoproto.nullable) = false];
+  bytes                     hash                 = 5;
+  int64                     height               = 6;
+  google.protobuf.Timestamp time                 = 7 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  // address of the public key of the node receiving this PrepareProposal request.
+  bytes proposer_address     = 8;
```

Comment:
This seems a little unnecessary but It's fine I guess

Reply:
Wait. If we remove this field from here, we may not be able to _immediately_ execute the block upon `PrepareProposal` or `ProcessProposal` (this field is present there as well), even if we are in next-block execution.
One of the main use cases of ABCI++ (even in next-block execution) is the possibility of executing the block at `PrepareProposal`/`ProcessProposal` in the same conditions as `FinalizeBlock` so that:
1. any invalid tx is spotted and excluded from the block
2. in the normal case, when we get the `FinalizeBlock` we just need to apply the corresponding candidate state, without any processing.

Also, regarding 1. above, what tells us that the proposer information may not be used for deciding whether a particular tx is valid or not?

Reply:
> Also, regarding 1. above, what tells us that the proposer information may not be used for deciding whether a particular tx is valid or not?

I'm not sure I can think of a case where that would be true yet, but perhaps we'd want to allow it. Either way, I've added it for parity with the other calls so that the block can be immediately executed.

@cmwaters let me know if there are other changes or kindly approve the PR :pray: